Core/clim-core/presentations/presentation-types.lisp

```diff
@@ -850,3 +850,91 @@ suitable for SUPER-NAME"))
                                   `(,name-and-params ,@needed-options)
                                   name-and-params))))
       name-and-params)))
+
+;;; This function is used to determine the CLIM presentation name from
+;;; the Common Lisp object.
+;;;
+;;; Not defined as a generic function, but what the hell.
+(defgeneric presentation-type-of (object))
```

Comment:
I"ve added a separate entry to decls.lisp, but it is the function declaration:
(declfun presentation-type-of (object))
let's keep the fact that it is a generic function as the implementation quirk, not as a promise

Reply:
Why is `defgeneric` in `decls.lisp` a promise? Wouldn't including the comment make the fact that the spec doesn't mandate a generic function clear?

The advantage of defining the generic function in `decls.lisp` is that it automatically precedes all method definitions.

Reply:
imo it encourages writing methods in arbitrary places while having it locally in one file entitles us to say, that this function is implemented in one place (and the fact that it is a generic function is an implementation detail which is local to said file).

Reply:
> imo defining the generic function globally encourages writing methods in arbitrary places

The ability (not necessarily encouragement) to do that is one of the points of generic functions.

> having it locally in one file entitles us to say, that this function is implemented in one place (and the fact that it is a generic function is an implementation detail which is local to said file).

Can't a `defgeneric` form in `decls.lisp` be an implementation detail as well? From my perspective it would be a detail of an implementation with the following advantages:
* One fewer forms (the `declfun` form)
* `defmethod` forms are guaranteed to follow the `defgeneric` form which avoids module and file ordering issues (and warnings due to those)

Reply:
The only benefit of putting defgeneric in decls is to guarantee, that a behavior which I don't want to happen (that is, that methods are defined elsewhere), will signal no warning.